blockly-core/core/ui/block_space/block_space.js

```diff
@@ -815,6 +818,20 @@ Blockly.BlockSpace.prototype.getScrollableSize = function(metrics) {
 };
 
 /**
+ * Returns the maximum possible scrolling offsets (+x right, +y down) for
+ * this BlockSpace.
+ * @returns {{x: number, y: number}}
+ */
+Blockly.BlockSpace.prototype.getMaxScrollOffsets = function() {
+  var metrics = this.getMetrics();
+  var blockSpaceSize = this.getScrollableSize(metrics);
+  return {
+    x: -blockSpaceSize.width + metrics.viewWidth,
```

Comment:
Much better. This actually made me realize many of the +/- described in the added method comments are actually flipped in this code path. Changing some values now to make this code path deal mostly with (+x right, +y down) numbers for readability.